Right to know - create new staging

jattree opened the pull request:
This is the pr for the stand alone staging rtk/ubuntu 22.04. Theres a lot so need eyes on this.

#217

Commit:
Add Ubuntu 22.04 support and staging.righttoknow.org.au configuration

- Add Ubuntu 22.04 (jammy) support for Right to Know playbook
- Update certbot role to use Ubuntu repos instead of PPA for 22.04
- Add --agree-tos flag for certbot certificate installation
- Add Ubuntu 22.04 specific package lists and passenger configuration
- Add boto3 support for Python 3 on Ubuntu 20.04+ (RDS info gathering)
- Add staging.righttoknow.org.au to inventory
- Add staging-specific group vars and host vars
- Update .gitignore to include both terraform files from rebase

Tested successfully deploying full Right to Know stack on Ubuntu 22.04 staging server.

Commit:
Revert .gitignore changes - vault files should remain as symlinks for team

Br3nda commented on group_vars/righttoknow-staging.yml:
```diff
@@ -0,0 +1,53 @@
+certbot_webserver: nginx
+certbot_webroot: /usr/share/nginx/html
+certbot_domains:
+  - staging.righttoknow.org.au
+  - staging-test.righttoknow.org.au
+  - www.staging-test.righttoknow.org.au
```
:sparkles:

Br3nda reviewed (approved):
looks good

benrfairless commented:
I've added some extra comments.

There's going to be a need to deploy different versions of various templates (general.yml, storage.yml etc).

One thing before I forget - we also need to make sure the MX records are set up properly and that Google Apps server is configured. @jamezpolley can you look at that?

jamezpolley commented on group_vars/righttoknow-staging.yml:
```diff
@@ -0,0 +1,53 @@
+cron_enabled: false
+
+# Staging-specific settings
+rails_env: staging
```
Looking at https://github.com/jamezpolley/infrastructure/blob/c4c60e641d5d4b374883cd4254356bf9682c27c9/README.md#L306 - all three environments are set to production. This confuses me.

Br3nda replied:
it's really just telling Rails what mode to be in. 

- Development (reloading files everytime your editor saves them). 
- Test (tear down and created all the time). 
- Production (real websites like production and staging)

jamezpolley commented:
Oh hey, this must be why you were asking me about mail routes @benrfairless . I'll look out for some regexes from you soon!

jamezpolley reviewed (commented):
Not approving as I think the rails_env needs to be updated? But also not requesting changes so this can be merged immediately if that turns out to be correct already

Commit:
Fix rails_env and add righttoknow_domain for staging

- Change rails_env from 'staging' to 'production' in both config files
  (Both staging and production run in Rails production mode)
- Add righttoknow_domain variable needed by certbot role for SSL cert generation
- Update comments for clarity

Addresses review feedback from @Br3nda

jattree reviewed (commented):
Updated to address @Br3nda's feedback on rails_env. Also added righttoknow_domain variable which is required by the certbot role for SSL certificate generation.

benrfairless commented on host_vars/staging.righttoknow.org.au.yml:
```diff
@@ -0,0 +1,16 @@
+test_domain: staging-test.righttoknow.org.au
+
+# Disable cron and backups for staging
+cron_enabled: false
```
Cron still needs to run in staging for some Alaveteli tasks

Br3nda replied:
we should build with cron off, so it doesn't pull from mailboxes right away

Br3nda commented on group_vars/righttoknow-staging.yml:
```diff
@@ -0,0 +1,53 @@
+
+# You might want different domains for staging
+domain: staging.righttoknow.org.au
+test_domain: staging-test.righttoknow.org.au
```
i think we don't need staging-test

jattree committed:
Update group_vars/righttoknow-staging.yml

Co-authored-by: Brenda Wallace <brenda@wallace.net.nz>

benrfairless committed:
Comment out unique Ansible connection settings for staging environment

benrfairless committed:
Add installation of ACL package so that Ansible can work properly

benrfairless committed:
Add condition for Stouts.backup role to execute only on bionic distribution

benrfairless committed:
Remove references to staging-test.righttoknow.org.au from Ansible and Terraform

benrfairless committed:
Add SPF and MX records for staging environment in DNS configuration
